tests/test-rust-setup.el

```diff
@@ -38,97 +38,147 @@
 (defun crate-file (file-name)
   (expand-file-name file-name "tests/test-crate"))
 
+(defun crate-with-features-file (file-name)
+  (expand-file-name file-name "tests/crate-with-features"))
+
 (defun lib-crate-file (file-name)
   (expand-file-name file-name "tests/custom-lib-target"))
 
 (defun build-script-crate-file (file-name)
   (expand-file-name file-name "tests/build-script-test"))
 
+(defun cdrassoc (sym alist) (cdr (assoc sym alist)))
+
+(defun get-cargo-version ()
+  (let ((cargo (funcall flycheck-executable-find "cargo")))
+    (with-temp-buffer
+      (call-process cargo nil '(t nil) nil "--version")
+      (buffer-string)
+      )))
+
+(defun cargo-version ()
+  (let* ((version-string-parts (split-string (get-cargo-version)))
+         (version (cadr version-string-parts))
+         (version-parts (split-string version "-")))
+    version-parts))
+
+(defun cargo-older-than-1.29-nightly ()
+  (let* ((version-parts (cargo-version))
+         (version (car version-parts))
+         (channel (cadr version-parts)))
+    (if (version< version "1.29")
+        t
+      (and (version= version "1.29") (string= channel "beta")))))
```

Comment:
Do we actually know which version `required-features` will be included in?  If it's `1.30`, then we don't need to check for the channel, right?

Reply:
Current `beta` has `cargo 1.29.0-beta (<sha> 2018-08-05)` (or some such). Nighty should get this feature in a week at most, and the version will be `1.29.0-nightly (<sha> 2018-08-21)`. We can just test for `1.30`, but then we will disable test for nightly `1.29` even though cargo will have new feature.

Not a big deal, I guess, and will help make code a bit nicer.

Reply:
Ah sure, let's test with 1.29-nightly.  No big deal.